scripts/hostcfgd

```diff
@@ -1249,6 +1256,97 @@ class PasswHardening(object):
         # set new Password Hardening policies.
         self.set_passw_hardening_policies(passw_policies)
 
+class SshServer(object):
+    def __init__(self):
+        self.policies = {}
+
+    def load(self, policies_conf):
+        for row in policies_conf:
+            self.policies_update(row, policies_conf[row], modify_conf=False)
+
+        self.modify_conf_file()
+
+    def modify_conf_file(self):
+        ssh_policies = {}
+        ssh_policies.update(self.policies)
+
+        # set new SSH server policies.
+        self.set_policies(ssh_policies)
+
+    def policies_update(self, key, data, modify_conf=True):
+        syslog.syslog(syslog.LOG_DEBUG, "ssh_policies_update - key: {}".format(key))
+        syslog.syslog(syslog.LOG_DEBUG, "ssh_policies_update - data: {}".format(data))
+
+        if data == {}:
+            self.policies = {}
+        else:
+            if 'ports' in data:
+                data['ports'] = data['ports'].split(',')
+            if key == 'POLICIES':
+                self.policies = data
+
+        if modify_conf:
+            self.modify_conf_file()
+
+    # return first line apperience of pattern - else return number of lines in the file
+    def get_line_num_of_pattern(self, pattern, file_path, find_commented=False):
+        syslog.syslog(syslog.LOG_DEBUG, "looking for pattern {} line in file {}".format(pattern, file_path))
+        return_value = 0
+        with open(file_path, 'r') as f:
+            for (i, line) in enumerate(f):
+                if re.match(pattern, line):
+                    syslog.syslog(syslog.LOG_DEBUG, "found pattern {} in line {}".format(pattern, str(i)))
+                    return i + 1
+                if find_commented and re.match('#' + pattern, line):
+                    syslog.syslog(syslog.LOG_DEBUG, "found pattern {} in line {}".format('#' + pattern, str(i)))
+                    return i + 1
+                return_value = i
+        return return_value
+
+    def handle_ports_set(self, values_list):
+        if not len(values_list) > 0:
+            return True
+        key='ports'
+        for port_num in values_list:
+            if isinstance(port_num, int):
+                syslog.syslog(syslog.LOG_ERR, "port num value {} in wrong format".format(port_num))
+                return False
+            if int(port_num) < SSH_MIN_VALUES[key] or SSH_MAX_VALUES[key] < int(port_num):
+                syslog.syslog(syslog.LOG_ERR, "Ssh {} {} out of range".format('port', port_num))
+                return False
+        port_line_num = self.get_line_num_of_pattern("Port", SSH_CONFG, True)
+        modify_single_file_inplace(SSH_CONFG, ['-E', "/^(#)?Port [0-9]+$/d"])
```

Comment:
>SSH_CONFG [](http://example.com/codeflow?start=35&length=9)

Modifying SSH_CONFG is dangerous.
`ssh -T` could be used to validate the config file. Could you leverage this capability?

Reply:
Good catch! In any cases, modified config without restarting service is dangerous. Including exception handling.

Reply:
@liuh-80  this is not the case - sshd_config can have all configurations empty and it will take its default ones anyway
@qiluo-msft I will add verification with sshd -T?

Reply:
@ycoheNvidia , I suggest modify/verify on tmp file, and rename the verified tmp file to config file:

admin@vlab-01:~$ sudo sshd -T  /tmp/sshd_config
Extra argument /tmp/sshd_config.